### ISSUE TITLE
Subclassing within UDP recommended?

### ISSUE THREAD

**Issue by ckaldemeyer** [esa/pagmo2#176]
Hi everybody,

I have a probably trivial question concerning the UDP which is triggered performance-wise.

Let's say I do the following to access common functionalities in different UDPs:

    class Base(object):

        def __init__(self, name):
            self.name = name

        def getName(self):
            return self.name

    class UDP(Base):

        def fitness(self, x):
            f1 = x[0] *2
            f2 = x[0] *3

            return [f1, f2]

        def get_nobj(self):
            return 2

        def get_bounds(self):
            return ([0]*1, [100]*1)

    if __name__ == '__main__':

        My_UDP = UDP(name='foo')
        print(My_UDP.name)

Does this have any performance disadvantages over passing an object e.g. as a class attribute into the UDP or accessing an "external" object from within the UDP except within the instantiation of the first generation individuals?

In other words: Are the UDP objects only created once and then only methods (fitness, ...) called in following generations?

Thanks in advance!

Cheers
Cord

**Comment by darioizzo** [comment]
Yes, the UDP are copied in the pg.problem object and then only their methods are called.

Please consider posting and following our gitter channel instead of opening tickets on github ... 
Gitter channel: https://gitter.im/pagmo2/Lobby
